janus_core/helpers/correlator.py

```diff
@@ -0,0 +1,409 @@
+                self._accumulator[block, 1] / self._window,
+                block + 1,
+            )
+            self._accumulator[block, :] = 0.0
+            self._count_accumulated[block] = 0
+
+        i = self._shift_index[block]
+        if block == 0:
+            j = i
+            for point in range(self._points):
+                if self._shifts_valid(block, i, j):
+                    self._correlation[block, point] += (
+                        self._shift[block, i, 0] * self._shift[block, j, 1]
+                    )
+                    self._count_correlated[block, point] += 1
+                j -= 1
+                if j < 0:
+                    j += self._points
+        else:
+            for point in range(self._min_dist, self._points):
+                if j < 0:
+                    j = j + self._points
+                if self._shifts_valid(block, i, j):
+                    self._correlation[block, point] += (
+                        self._shift[block, i, 0] * self._shift[block, j, 1]
+                    )
+                    self._count_correlated[block, point] += 1
+                j = j - 1
+        self._shift_index[block] = (self._shift_index[block] + 1) % self._points
+
+    def _shifts_valid(self, block: int, p_i: int, p_j: int) -> bool:
+        """
+        True if the shift registers have data.
+
+        Parameters
+        ----------
+        block : int
+            Block to check the shift register of.
+        p_i : int
+            Index i in the shift (left correland).
+        p_j : int
+            Index j in the shift (right correland).
+
+        Returns
+        -------
+        bool
+            Whether the shift indices have data.
+        """
+        return self._shift_not_null[block, p_i] and self._shift_not_null[block, p_j]
+
+    def get(self) -> tuple[Iterable[float], Iterable[float]]:
+        """
+        Obtain the correlation and lag times.
+
+        Returns
+        -------
+        tuple[Iterable[float], Iterable[float]]
+            The correlation and lags.
+        """
+        correlation = np.zeros(self._points * self._blocks)
+        lags = np.zeros(self._points * self._blocks)
+
+        lag = 0
+        for i in range(self._points):
+            if self._count_correlated[0, i] > 0:
+                correlation[lag] = (
+                    self._correlation[0, i] / self._count_correlated[0, i]
+                )
+                lags[lag] = i
+                lag += 1
+        for k in range(1, self._max_block_used):
+            for i in range(self._min_dist, self._points):
+                if self._count_correlated[k, i] > 0:
+                    correlation[lag] = (
+                        self._correlation[k, i] / self._count_correlated[k, i]
+                    )
+                    lags[lag] = float(i) * float(self._window) ** k
+                    lag += 1
+        return (correlation[0:lag], lags[0:lag])
+
+
+# pylint: disable=R0903
```

Comment:
Preferably use the long names of these issues so it's possible to see what you're trying to do without looking them up.
```suggestion
# pylint: disable=too-few-public-methods
```

Reply:
good point, didn't realise that could be done for some reason, I'll add these in